### DISCUSSION TITLE
Data Screening - InstrumenrResample

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#215] [Q&A]
Hello,

I am trying to do some trials on data screening in instrument files (orbit files for example). For example, I want to take out 10 minutes at the beginning and at the end of a segment. And then eventually do this for all the daily segments through a loop.
I have been using the instrumentResample script with the “Exclude function”, but I am having some problems understanding the different inputs in the script. Maybe there is another function that is better suited for this purpose like IntrumentRemoveEpochsBTimes.

First, I did some trials with a one segment setup.
The description states that a time series is first created, what variable would I put into the cells timeStart and timeEnd as theses variables come from the instrument file (basically the time column). That seemed to work as I did not get any errors, both variables are defined as global variables.
Then I would think the same variable goes into the timeStart and timeEnd under the “timeSeriesExclude” part? And then define the exclude margin as the 10 minutes or 600/89400? I did this also for a one segment run but got an empty file as output. I tried again with adding / subtracting the 10 min in timeStart / timeEnd and still got an empty file. According to the description the timeStart/timeEnd variables are the same in the timeseries and timeSeriesExclude part?

In a second step if I want to do this over all the daily segments, I will need to define timeStart and timeEnd as variable that is updated every time a new segment is run. And I am not sure how that is then defined in the instrumentResample script as the data for the variables come from the instrument files. 

I have included a print screen of a simple one segment script.

Thanks very much for the help.

BR
Jean
![Capture](https://github.com/groops-devs/groops/assets/100558965/aa6c6fb8-53b8-4a2c-84ce-7b9216956bbe)

**Comment by Bemtgenj** [comment]
Update

instrumentRemoveEpochsbyTime might be a better option to remove data at the beginning (and end of the segment). i think i would have to define a Start time and an End time > than the 10 minutes to take the first 10 minutes out. And then in a similar way use InstrumentRemoveEpochsbyTime again for the end of the segment where timeStart starts more than 10 minutes before the end and timeEnd is the end of the segment?

Then that still leaves the point open with redefining the beginning and end times of the segment every time a new segment is running through the different segment loops.

Cheers
Jean

**Comment by tmayerguerr** [comment]
Dear Jean

You might define the segements/arcs with [InstrumentSynchronize](https://groops-devs.github.io/groops/html/InstrumentSynchronize.html). It started a new arc after a `minGap` defined gap. In the next step I would use `removalCriteria` in [InstrumentArcCalculate](https://groops-devs.github.io/groops/html/InstrumentArcCalculate.html). There are variables `index` (data number, starting with zero) and `data0Count`  for each arc, e.g.
```removalCriteria = (index < 10) || (index > data0Count-10)```
to remove the first and last 10 epochs of an arc.

regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

Txs for the quick feedback. I will try what you mentioned above.

Cheers
Jean
